rust-libretro/src/environment.rs

```diff
@@ -959,14 +1024,15 @@ pub unsafe fn set_serialization_quirks(
 /// being used.
 #[proc::context(GenericContext)]
 #[proc::unstable(feature = "env-commands")]
-pub unsafe fn set_hw_shared_context(callback: retro_environment_t) -> bool {
```

Comment:
`RETRO_ENVIRONMENT_SET_HW_SHARED_CONTEXT` can return `false` to indicate that shared hardware contexts aren't supported. I'd argue that asking this question and getting a "no" isn't an error.

Reply:
Since this function doesn't take any data and just returns `true` or `false` for the callback, we end up with
`if ctx.set_hw_shared_context().is_ok() { ... }` vs. `if let Ok(true) = ctx.set_hw_shared_context() { ... }`
of those two cases I prefer the first, both give the same knowledge: Either the call succeeded or it failed.
The former knows two error cases: The environment call function pointer was a null pointer or the callback returned `false` (which is the general error case for the environment calls).
The latter replaces the second error case with `Ok(false)` and requires more work to check if the call succeeded.